pandas_gbq/gbq.py

```diff
@@ -574,15 +576,26 @@ def verify_schema(self, dataset_id, table_id, schema):
                               'type': field_remote['type']}
                              for field_remote in remote_schema['fields']]
 
-            fields_remote = set([json.dumps(field_remote)
-                                 for field_remote in remote_fields])
-            fields_local = set(json.dumps(field_local)
-                               for field_local in schema['fields'])
-
-            return fields_remote == fields_local
+            return remote_fields
         except HttpError as ex:
             self.process_http_error(ex)
 
+    def verify_schema(self, dataset_id, table_id, schema):
+        fields_remote = set([json.dumps(field)
+                             for field in self.schema(dataset_id, table_id)])
+        fields_local = set(json.dumps(field_local)
+                           for field_local in schema['fields'])
+
+        return fields_remote == fields_local
+
+    def schema_is_subset(self, dataset_id, table_id, schema):
+        fields_remote = set([json.dumps(field)
+                             for field in self.schema(dataset_id, table_id)])
+        fields_local = set(json.dumps(field_local)
+                           for field_local in schema['fields'])
+
```

Comment:
these routines should prob just take the result of ``self.schema(dataset_id, table_id)``, yes?